src/zarr/v3/group.py

```diff
@@ -302,22 +303,26 @@ async def members(self) -> AsyncGenerator[tuple[str, AsyncArray | AsyncGroup], N
             )
 
             raise ValueError(msg)
-        subkeys = await self.store_path.store.list_dir(self.store_path.path)
-        # would be nice to make these special keys accessible programmatically,
-        # and scoped to specific zarr versions
-        subkeys_filtered = filter(lambda v: v not in ("zarr.json", ".zgroup", ".zattrs"), subkeys)
-        # is there a better way to schedule this?
-        for subkey in subkeys_filtered:
-            try:
-                yield (subkey, await self.getitem(subkey))
-            except KeyError:
-                # keyerror is raised when `subkey` names an object (in the object storage sense),
-                # as opposed to a prefix, in the store under the prefix associated with this group
-                # in which case `subkey` cannot be the name of a sub-array or sub-group.
-                logger.warning(
-                    "Object at %s is not recognized as a component of a Zarr hierarchy.", subkey
-                )
-                pass
+
+        async for key in self.store_path.store.list_dir(self.store_path.path):
+            # these keys are not valid child names so we make sure to skip them
+            # TODO: it would be nice to make these special keys accessible programmatically,
+            # and scoped to specific zarr versions
+            if key not in ("zarr.json", ".zgroup", ".zattrs"):
+                try:
+                    # TODO: performance optimization -- load children concurrently   
+                    child = await self.getitem(key)
+                    # keyerror is raised when `key``names an object in the store
+                    # in which case `subkey` cannot be the name of a sub-array or sub-group.
+                    yield key, child
+                except KeyError:
+                    # keyerror is raised when `subkey` names an object (in the object storage sense),
+                    # as opposed to a prefix, in the store under the prefix associated with this group
+                    # in which case `subkey` cannot be the name of a sub-array or sub-group.
+                    logger.warning(
```

Comment:
Are both of these comments necessary, or can we just use the second one (after changing `subkey` to `key`)?

Reply:
```suggestion
                    yield key, child
                except KeyError:
                    # keyerror is raised when `key` names an object (in the object storage sense),
                    # as opposed to a prefix, in the store under the prefix associated with this group.
                    # In this case, `key` cannot be the name of a sub-array or sub-group.
                    logger.warning(
```